streams/src/test/java/org/apache/kafka/streams/kstream/KStreamBuilderTest.java

```diff
@@ -150,4 +153,34 @@ public void shouldThrowExceptionWhenTopicNamesAreNull() throws Exception {
         new KStreamBuilder().stream(Serdes.String(), Serdes.String(), null, null);
     }
 
+    @Test
+    public void shouldNotModifyStateStoreSourceIfItExistsAndAnotherProcessorConnectingToIt() throws Exception {
```

Comment:
nit: Can we just merge these two test into one? I feel the second is almost a super set coverage of the first.